test/test_util.jl

```diff
@@ -144,7 +162,7 @@ end
 function Base.isapprox(d_ad::DNE, d_fd; kwargs...)
     error("Tried to differentiate w.r.t. a DNE")
 end
-function Base.isapprox(d_ad::Thunk, d_fd; kwargs...)
+function Base.isapprox(d_ad::Union{Thunk,One,Zero}, d_fd; kwargs...)
```

Comment:
```suggestion
function Base.isapprox(d_ad::AbstractDifferential, d_fd; kwargs...)
```

In general `extern` is what one does to an `AbstractDifferential` to get a numeric value.
And we have covered the special cases.
And `Casted` is its own ball of fun #12  that can be worried about another day